snapcraft/internal/lifecycle.py

```diff
@@ -108,7 +109,11 @@ def execute(step, project_options, part_names=None):
     :returns: A dict with the snap name, version, type and architectures.
     """
     config = snapcraft.internal.load_config(project_options)
-    repo.Repo.install_build_packages(config.build_tools)
+    installed_packages = repo.Repo.install_build_packages(
+        config.build_tools)
+    os.makedirs('snap/.snapcraft', exist_ok=True)
+    with open('snap/.snapcraft/state', 'w') as f:
```

Comment:
Please kindly use `os.path.join` here. You're a wonderful person, and your work is appreciated.